extension/src/Documentation.ts

```diff
@@ -123,7 +123,7 @@ export async function generateExternalDocumentation(
 
   const publicObjects = objects.filter(
     (obj) =>
-      ([
+      (([
```

Comment:
Maybe not this time but for the future.

Suggestion 1:  move ALObjectType array to a separate inst.
Suggestion 2: Add braces `{}` to arrow function, declare all boolean checks as const and add a compact return statement in the end. 

eg. (after piecing this together I'm not sure it's actually better. But I think this would be easier to debug :wink:
```javascript
// Completely untested!
  const publicObjects = objects.filter((obj) => {
    const exportObjType = [
      ALObjectType.codeunit,
      ALObjectType.interface,
      ALObjectType.permissionSet,
      ALObjectType.query,
      ALObjectType.report,
      ALObjectType.reportExtension,
      ALObjectType.table,
      ALObjectType.tableExtension,
      ALObjectType.xmlPort,
    ].includes(obj.getObjectType());

    const isNormalSubType = obj.subtype === ALCodeunitSubtype.normal;
    const hasPublicControls =
      obj.controls.filter(
        (proc) =>
          (proc.type === ALControlType.procedure &&
            (proc as ALProcedure).access === ALAccessModifier.public) ||
          (proc as ALProcedure).event
      ).length > 0;
    const tableOrTableExt =
      settings.includeTablesAndFieldsInDocs &&
      obj.publicAccess &&
      [ALObjectType.table, ALObjectType.tableExtension].includes(
        obj.getObjectType()
      );
    const publicAccessExtensibleEnum =
      (obj.publicAccess &&
        obj.getObjectType() === ALObjectType.enum &&
        obj.getProperty(ALPropertyType.extensible, false)) ||
      ([ALObjectType.page, ALObjectType.pageExtension].includes(
        obj.getObjectType()
      ) &&
        !obj.apiObject);
    return (
      (exportObjType &&
        obj.publicAccess &&
        isNormalSubType &&
        hasPublicControls) ||
      tableOrTableExt ||
      publicAccessExtensibleEnum
    );
  });
```
Some of these  could probably be extracted to getters or function on `ALObject`.

Reply:
Yes, I agree. Something has to be done too this monster. It has grown beyond control!😂

But I think that's a challenge for another day!
I also think I need to add more AL objects to our test project, to cover all different cases here..